bug: Panic after weeks of running

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Current Behavior

The coder docker container exited due to internal panic.

### Relevant Log Output

```shell
panic: length < 0

goroutine 248430109 [running]:
gvisor.dev/gvisor/pkg/tcpip/stack.PacketData.CapLength(...)
        /home/runner/go/pkg/mod/gvisor.dev/gvisor@v0.0.0-20240509041132-65b30f7869dc/pkg/tcpip/stack/packet_buffer.go:581
gvisor.dev/gvisor/pkg/tcpip/transport/tcp.(*sender).splitSeg(0xc01bc86008, 0xc03d0ac800, 0xfe32de9401bff440)
        /home/runner/go/pkg/mod/gvisor.dev/gvisor@v0.0.0-20240509041132-65b30f7869dc/pkg/tcpip/transport/tcp/snd.go:639 +0x185
gvisor.dev/gvisor/pkg/tcpip/transport/tcp.(*sender).maybeSendSegment(0xc01bc86008, 0xc03d0ac800, 0xfe32de9401bff440?, 0xbcfd9b78?)
        /home/runner/go/pkg/mod/gvisor.dev/gvisor@v0.0.0-20240509041132-65b30f7869dc/pkg/tcpip/transport/tcp/snd.go:908 +0x171
gvisor.dev/gvisor/pkg/tcpip/transport/tcp.(*sender).sendData(0xc01bc86008)
        /home/runner/go/pkg/mod/gvisor.dev/gvisor@v0.0.0-20240509041132-65b30f7869dc/pkg/tcpip/transport/tcp/snd.go:1031 +0x1b3
gvisor.dev/gvisor/pkg/tcpip/transport/tcp.(*Endpoint).sendData(0xc024f88e08?, 0x0?)
        /home/runner/go/pkg/mod/gvisor.dev/gvisor@v0.0.0-20240509041132-65b30f7869dc/pkg/tcpip/transport/tcp/connect.go:1007 +0x45
gvisor.dev/gvisor/pkg/tcpip/transport/tcp.(*Endpoint).Write(0xc024f88e08, {0xd91af08, 0xc0592850b0}, {0x0, 0x0, 0x0, 0x0, {0x0, 0x0, 0x0, ...}})
        /home/runner/go/pkg/mod/gvisor.dev/gvisor@v0.0.0-20240509041132-65b30f7869dc/pkg/tcpip/transport/tcp/endpoint.go:1663 +0x127
gvisor.dev/gvisor/pkg/tcpip/adapters/gonet.(*TCPConn).Write(0xc019114f60, {0xc02d1e6000, 0xc, 0x8000})
        /home/runner/go/pkg/mod/gvisor.dev/gvisor@v0.0.0-20240509041132-65b30f7869dc/pkg/tcpip/adapters/gonet/gonet.go:389 +0x256
io.copyBuffer({0xd8edf40, 0xc019114f60}, {0x7b89fc10a9d8, 0xc0309ca978}, {0x0, 0x0, 0x0})
        /opt/hostedtoolcache/go/1.24.6/x64/src/io/io.go:431 +0x1d7
io.Copy(...)
/opt/hostedtoolcache/go/1.24.6/x64/src/io/io.go:388
github.com/coder/coder/v2/agent/agentssh.Bicopy.func2({0x7b89fc10a9b0?, 0xc019114f60?}, {0x7b89fc10a9d8?, 0xc0309ca978?})
        /home/runner/work/coder/coder/agent/agentssh/bicopy.go:29 +0x8b
created by github.com/coder/coder/v2/agent/agentssh.Bicopy in goroutine 248417584
        /home/runner/work/coder/coder/agent/agentssh/bicopy.go:34 +0x2f8
```

### Expected Behavior

Coder should not panic without any reason

### Steps to Reproduce

Not certain

### Environment

- Host OS: Ubuntu 22.04
- Coder version: v2.26.3


### Additional Context

The issue is new (previously worked fine)

--- a/go.mod
+++ b/go.mod
@@ -43,6 +43,10 @@ replace tailscale.com => github.com/coder/tailscale v1.1.1-0.20260306035934-af5c
 // 2. update to the latest gVisor
 replace github.com/tailscale/wireguard-go => github.com/coder/wireguard-go v0.0.0-20260113101225-9b7a56210e49
 
+// We use a fork to fix an integer overflow issue that causes occasional crashes in workspace agents.
+// See https://github.com/coder/coder/issues/20885
+replace gvisor.dev => github.com/coder/gvisor v0.0.0-20260313164934-7a658db7b714
+
 // Switch to our fork that imports fixes from http://github.com/tailscale/ssh.
 // See: https://github.com/coder/coder/issues/3371
 //

